test/e2e_new/broker_trigger_sink_test.go

```diff
@@ -0,0 +1,157 @@
+	trigger2Name := feature.MakeRandomK8sName("trigger-2")
+	sink := feature.MakeRandomK8sName("sink")
+	verifyMessagesJobName := feature.MakeRandomK8sName("verify-messages-job")
+
+	topic := kafka.BrokerTopic(brokerreconciler.TopicPrefix, &eventingv1.Broker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      brokerName,
+			Namespace: env.Namespace(),
+		},
+	})
+
+	trigger1FilterType := "trigger-1"
+	trigger2FilterType := "trigger-2"
+
+	eventIdTrigger1 := uuid.New().String()
+	eventTrigger1 := test.MinEvent()
+	eventTrigger1.SetID(eventIdTrigger1)
+	eventTrigger1.SetType(trigger1FilterType)
+
+	eventIdTrigger2 := uuid.New().String()
+	eventTrigger2 := test.MinEvent()
+	eventTrigger2.SetID(eventIdTrigger2)
+	eventTrigger2.SetType(trigger2FilterType)
+
+	f := feature.NewFeatureNamed("Trigger with KafkaSink")
+
+	f.Setup("Create broker config", brokerconfigmap.Install(
+		brokerConfig,
+		brokerconfigmap.WithBootstrapServer(testpkg.BootstrapServersPlaintext),
+		brokerconfigmap.WithNumPartitions(testpkg.NumPartitions),
+		brokerconfigmap.WithReplicationFactor(testpkg.ReplicationFactor),
+	))
+
+	f.Setup("Install broker", broker.Install(brokerName,
+		append(
+			broker.WithEnvConfig(),
+			broker.WithConfig(brokerConfig))...,
+	))
+	f.Setup("Broker is ready", broker.IsReady(brokerName))
+	f.Setup("Topic is ready", kafkatopic.IsReady(topic))
+
+	f.Setup("Install kafkasink", kafkasink.Install(sink, topic, testpkg.BootstrapServersPlaintextArr,
+		kafkasink.WithNumPartitions(10),
+		kafkasink.WithReplicationFactor(1)))
+	f.Setup("KafkaSink is ready", kafkasink.IsReady(sink))
+
+	f.Setup("Create trigger 1", trigger.Install(trigger1Name, brokerName,
+		trigger.WithSubscriber(kafkasink.AsKReference(sink, env.Namespace()), ""),
+		trigger.WithFilter(map[string]string{
+			"type": trigger1FilterType,
+		})))
+	f.Setup("Trigger 1 is ready", trigger.IsReady(trigger1Name))
+
+	f.Setup("Create trigger 2", trigger.Install(trigger2Name, brokerName,
+		trigger.WithSubscriber(kafkasink.AsKReference(sink, env.Namespace()), ""),
+		trigger.WithFilter(map[string]string{
+			"type": trigger2FilterType,
+		})))
+	f.Setup("Trigger 2 is ready", trigger.IsReady(trigger2Name))
+
+	f.Requirement("Send event for trigger 1", eventshub.Install(feature.MakeRandomK8sName("sender-1"),
+		eventshub.InputEvent(eventTrigger1),
+		eventshub.StartSenderToResource(broker.GVR(), brokerName)))
+	f.Requirement("Send event for trigger 2", eventshub.Install(feature.MakeRandomK8sName("sender-2"),
+		eventshub.InputEvent(eventTrigger2),
+		eventshub.StartSenderToResource(broker.GVR(), brokerName)))
+
+	f.Requirement("Install verify-messages job", job.Install(verifyMessagesJobName, kafkaConsumerImage,
+		job.WithEnvs(map[string]string{
+			"BOOTSTRAP_SERVERS": testpkg.BootstrapServersPlaintext,
+			"TOPIC":             topic,
+			"IDS":               fmt.Sprintf("%s,%s", eventIdTrigger1, eventIdTrigger2),
+			"CONTENT_MODE":      "structured",
```

Comment:
I just noticed this and affects the existing test as well, so we can ignore it for the purpose of this PR.

Since we are not setting the content mode when we install the sink [1] I would expect this to be "binary" since that's the default content mode?

https://github.com/knative-sandbox/eventing-kafka-broker/blob/9459144e95bd13df164340970a96b58a2761daa4/control-plane/pkg/apis/eventing/v1alpha1/kafka_sink_defaults.go#L28-L32

so:
- that verification is not happening, or
- there is a bug in the sink implementation for changing content mode

The existing test is doing the same thing and it passes as well

[1]
```golang
	f.Setup("Install kafkasink", kafkasink.Install(sink, topic, testpkg.BootstrapServersPlaintextArr,
		kafkasink.WithNumPartitions(10),
		kafkasink.WithReplicationFactor(1)))
```